sdk/core/azure-core-amqp/inc/azure/core/amqp.hpp

```diff
@@ -6,13 +6,17 @@
 #include "azure/core/amqp/claims_based_security.hpp"
 #include "azure/core/amqp/common/async_operation_queue.hpp"
 #include "azure/core/amqp/common/completion_operation.hpp"
+#include "azure/core/amqp/common/global_state.hpp"
```

Comment:
Except for amqp_error.hpp, and endpoint.hpp, none these files should be in the inclusive file. They contain implementation details of AMQP and cannot be meaningfully consumed outside the AMQP codebase.

This is *especially* true for doxygen_pragma which is intended to hide types from doxygen.

Technically it should be in azure core but I didn't want to introduce it into azure core because AMQP is the only package currently which needs it.

Reply:
These files are available in intellisense once user types `#include <azure/core/amqp/common/`.

You are right, but then it means we should move them to `azure/core/amqp/common/internal/`. All other libraries follow this convention - I know Core and Storage.

Reply:
#5069